tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java

```diff
@@ -108,4 +113,61 @@ public MenuBar createFailingMenuBar() {
 
         return menuBar;
     }
+
+    @Test
+    public void testMemoryLeak() throws InterruptedException {
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
+            });
+            createMenuBarWithItemsStage();
+        });
+        memoryLatch.await();
+        assertFalse(failed.get());
+    }
+
+    public void createMenuBarWithItemsStage() {
+        final ArrayList<WeakReference<MenuItem>> uncollectedMenuItems = new ArrayList<>();
+
+        Stage stage = new Stage();
+        VBox root = new VBox();
+        final MenuBar menuBar = new MenuBar();
+        final Menu menu = new Menu("MyMenu");
+        menuBar.getMenus().add(menu);
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+        stage.requestFocus();
+        Thread t = new Thread(){
+            @Override public void run() {
+                for (int i = 0; i < 50; i++) {
+                    try {
+                        Thread.sleep(20);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    Platform.runLater(() -> {
+                        menu.getItems().clear();
+                        MenuItem menuItem = new MenuItem("MyItem");
+                        WeakReference wr = new WeakReference<>(menuItem);
+                        uncollectedMenuItems.add(wr);
+                        menu.getItems().add(menuItem);
+                    });
+                }
+                Platform.runLater( () -> {
+                    System.gc();
+                    uncollectedMenuItems.removeIf(ref -> ref.get() == null);
```

Comment:
This assumes that the references will all be collected by a single call to `System.gc` with no delay. While it worked on my system, it might be fragile. Have you considered using JMemoryBuddy? Or if that isn't suitable for some reason, maybe consider looping a few times with a sleep in between each call to gc?